### PR TITLE
[PFsense] Enhance PFsense Integration with DHCPv6 support

### PR DESCRIPTION
**Pull request by legoguy1000**
## What does this PR do?

Add DHCPv6 support

## Checklist

- [X] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [X] I have verified that all data streams collect metrics or logs.
- [X] I have added an entry to my package's `changelog.yml` file.
- [X] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ]

## Related issues

-

### CODE REVIEW

File: packages/pfsense/data_stream/log/elasticsearch/ingest_pipeline/default.yml [hunk 1]
```diff
@@ -61,7 +64,7 @@ processors:
       if: ctx.event.provider == 'charon'
   - pipeline:
       name: '{{ IngestPipeline "dhcp" }}'
-      if: ctx.event.provider == 'dhcpd'
+      if: '["dhcpd", "dhclient", "dhcp6c"].contains(ctx?.event?.provider)'
```
Comment (efd6) [inline packages/pfsense/data_stream/log/elasticsearch/ingest_pipeline/default.yml:67]: We know that `ctx` is non-null and also that `ctx.event` is non-null (due to `event.kind` and `event.original` above).

```suggestion
      if: '["dhcpd", "dhclient", "dhcp6c"].contains(ctx.event.provider)'
```

File: packages/pfsense/data_stream/log/elasticsearch/ingest_pipeline/default.yml [hunk 2]
```diff
@@ -75,7 +78,31 @@ processors:
       name: '{{ IngestPipeline "squid" }}'
       if: ctx.event.provider == 'squid'
   - drop:
-      if: '!["filterlog", "openvpn", "charon", "dhcpd", "unbound", "haproxy", "php-fpm", "squid"].contains(ctx?.event?.provider)'
+      if: '!["filterlog", "openvpn", "charon", "dhcpd", "dhclient", "dhcp6c", "unbound", "haproxy", "php-fpm", "squid"].contains(ctx?.event?.provider)'
+  - append:
+      field: event.category
+      value: network
+      if: "ctx?.network != null"
```
Comment (efd6) [inline packages/pfsense/data_stream/log/elasticsearch/ingest_pipeline/default.yml:85]:
```suggestion
      if: "ctx.network != null"
```

File: packages/pfsense/data_stream/log/elasticsearch/ingest_pipeline/default.yml [hunk 3]
```diff
@@ -75,7 +78,31 @@ processors:
...
+  - convert:
+      field: destination.address
+      target_field: destination.ip
+      type: ip
+      ignore_failure: true
+      ignore_missing: true
+  - set:
+      field: network.type
+      value: ipv6
+      if: 'ctx.source?.ip != null && ctx.source?.ip.contains(":")'
```
Comment (efd6) [inline packages/pfsense/data_stream/log/elasticsearch/ingest_pipeline/default.yml:101]:
```suggestion
      if: 'ctx.source?.ip != null && ctx.source.ip.contains(":")'
```

File: packages/pfsense/data_stream/log/elasticsearch/ingest_pipeline/default.yml [hunk 4]
```diff
@@ -75,7 +78,31 @@ processors:
...
+  - set:
+      field: network.type
+      value: ipv4
+      if: 'ctx.source?.ip != null && ctx.source?.ip.contains(".")'
```
Comment (efd6) [inline packages/pfsense/data_stream/log/elasticsearch/ingest_pipeline/default.yml:105]:
```suggestion
      if: 'ctx.source?.ip != null && ctx.source.ip.contains(".")'
```

File: packages/pfsense/data_stream/log/elasticsearch/ingest_pipeline/dhcp.yml [hunk 5]
```diff
@@ -4,7 +4,13 @@ processors:
   - grok:
       field: message
       patterns:
-        - '%{WORD:event.action} %{MIDDLE} via %{INTERFACE:observer.ingress.interface.name}' 
+        # - '%{DATA:message}/%{DATA:interface}/%{DATA:server.mac}/%{NOTSPACE:subnet}'
```
Comment (efd6) [inline packages/pfsense/data_stream/log/elasticsearch/ingest_pipeline/dhcp.yml:7]: Delete commented out line.

File: packages/pfsense/data_stream/log/elasticsearch/ingest_pipeline/dhcp.yml [hunk 6]
```diff
@@ -13,47 +19,36 @@ processors:
...
-      allow_duplicates: false
-  - append:
-      field: event.type
-      value: connection
-      allow_duplicates: false
-  - append:
-      field: event.type
-      value: protocol
-      allow_duplicates: false
+        DUID: '[0-9a-f]{2}(:[0-9a-f]{2})+'
```
Comment (efd6) [inline packages/pfsense/data_stream/log/elasticsearch/ingest_pipeline/dhcp.yml:22]: DUIDs are specified on top of UID which allows uppercase A-F.
```suggestion
        DUID: '(?i)[0-9a-f]{2}(:[0-9a-f]{2})+'
```

File: packages/pfsense/data_stream/log/elasticsearch/ingest_pipeline/dhcp.yml [hunk 7]
```diff
@@ -13,47 +19,36 @@ processors:
...
+        - protocol
+        - info
       allow_duplicates: false
   - set:
       field: network.protocol
       value: dhcp
+  - set:
+      field: network.protocol
+      value: dhcpv6
+      if: ctx.event?.provider == 'dhcp6c' || (ctx?.server?.address != null && ctx?.server?.address.contains(':')) || (ctx?.client?.address != null && ctx?.client?.address.contains(':'))
```
Comment (efd6) [inline packages/pfsense/data_stream/log/elasticsearch/ingest_pipeline/dhcp.yml:36]:
```suggestion
      if: ctx.event.provider == 'dhcp6c' || (ctx.server?.address != null && ctx.server.address.contains(':')) || (ctx.client?.address != null && ctx.client.address.contains(':'))
```

File: packages/pfsense/data_stream/log/elasticsearch/ingest_pipeline/dhcp.yml [hunk 8]
```diff
@@ -62,6 +57,15 @@ processors:
       pattern: '[:]'
       replacement: '-'
       ignore_missing: true
+  - lowercase:
+      field: _tmp.action
+      ignore_missing: true
+  - gsub:
+      field: _tmp.action
+      target_field: event.action
+      pattern: '[ ]'
```
Comment (efd6) [inline packages/pfsense/data_stream/log/elasticsearch/ingest_pipeline/dhcp.yml:66]: Does this need to be a char set?

File: packages/pfsense/data_stream/log/elasticsearch/ingest_pipeline/firewall.yml [hunk 9]
```diff
@@ -33,23 +33,11 @@ processors:
...
-    - set:
-        field: destination.ip
-        value: "{{destination.address}}"
-        ignore_empty_value: true
     - append:
         field: event.type
         value: connection
         allow_duplicates: false
-        if: ctx?.source?.ip != null && ctx?.destination?.ip != null
+        if: ctx?.source?.address != null && ctx?.destination?.address != null
```
Comment (efd6) [inline packages/pfsense/data_stream/log/elasticsearch/ingest_pipeline/firewall.yml:40]:
```suggestion
        if: ctx.source?.address != null && ctx.destination?.address != null
```

File: packages/pfsense/data_stream/log/elasticsearch/ingest_pipeline/haproxy.yml [hunk 10]
```diff
@@ -105,21 +73,14 @@ processors:
...
       if: "ctx?.haproxy?.mode == 'HTTP' || ctx?.haproxy?.http != null"
-  - append:
-      field: event.category
-      value: network
-      if: "ctx?.source.ip != null && ctx?.destination?.ip != null"
   - append:
       field: event.type
       value: connection
-      if: "ctx?.source.ip != null && ctx?.destination?.ip != null"
+      if: "ctx?.source.address != null && ctx?.destination?.address != null"
```
Comment (efd6) [inline packages/pfsense/data_stream/log/elasticsearch/ingest_pipeline/haproxy.yml:83]:
```suggestion
      if: "ctx.source.address != null && ctx.destination?.address != null"
```

Also query that we know `ctx.source` is non-null at this point? Should a null-safe selector be added to that access?

File: packages/pfsense/data_stream/log/elasticsearch/ingest_pipeline/dhcp.yml (hunk 6 above)
Comment (legoguy1000) [reply packages/pfsense/data_stream/log/elasticsearch/ingest_pipeline/dhcp.yml:22]: PFsense appear to be all lowercase DUIDs but i'll add it to be safe

**Comment by legoguy1000** [comment]
updated.  Removed all `ctx?.`

**Review by efd6 — APPROVED** [review]

**Pull request merged**